### PR TITLE
feat!: convert to typescript

### PR DESCRIPTION
**Pull request by achingbrain**
- Converts to typescript
- Only named exports
- No more CJS, only ESM
- Uses aegir for building/testing/publishing
- Publishes with provenance
- Publishes API docs to gh-pages branch
- Switches travis for GitHub CI
- Updates deps

Similar in scope to https://github.com/alanshaw/it-ws/pull/6 etc

### CODE REVIEW

**Comment by achingbrain** [comment]
CI hasn't run here because I can't push branches to this repo so the PR comes from a fork..

File: benchmark/memory.js [hunk 1]
```diff
@@ -12,7 +12,7 @@
 const fs = require('fs')
 const os = require('os')
 const toIterable = require('../')
-const pipe = require('it-pipe')
+import { pipe } from 'it-pipe'
```
Comment (alanshaw) [inline benchmark/memory.js:14]: This needs some more love? i.e. convert the `require`s

**Review by alanshaw — APPROVED** [review]
LGTM

File: benchmark/memory.js (hunk 1 above)
Comment (achingbrain) [reply benchmark/memory.js:14]: Done!

**Pull request merged**